lib/ruby/lib/alces/stack/boot/run.rb

```diff
@@ -37,7 +37,10 @@ def initialize(options={})
           @finder = Alces::Stack::Templater::Finder.new("#{ENV['alces_BASE']}/etc/templates/boot/", options[:template])
           @group = options[:group]
           @dry_run_flag = options[:dry_run_flag]
+          @permanent_boot_flag = options[:permanent_boot_flag]
```

Comment:
There are a lot of `options` now (and accompanying instance variables).  It might be worth considering extracting these into an options utility class, e.g.:

```
class Run
  def initialize(options)
    @options = Options.new(options)
  end

  class Options
    def initialize(options)
      @options = options
      # [...more as necessary...]
    end

    def template_parameters
      @template_parameters ||= {
        firstboot: true,
        permanentboot: permanent_boot_flag,
        kernelappendoptions: kernel_append.chomp
      }
    end

    def dry_run?
       !!dry_run_flag
    end

    def method_missing(s, *a, &b)
      if options.key?(s)
        options[s]
      else
        super
      end
    end
  end
end
```

With that in place you can move some of the logic into the `Options` class as method implementations and make the `Run` class a bit more readable.